crates/runtime/src/component/dataset.rs

```diff
@@ -344,6 +353,56 @@ impl Dataset {
         }
     }
 
+    /// For [`Dataset`]s where the path in the `from` field is a [`TableReference`], parse and return the [`TableReference`].
+    ///
+    ///
+    #[must_use]
+    pub fn table_reference_path(
+        &self,
+        case_sensitive: bool,
+        dialect: Option<&dyn Dialect>,
+    ) -> Result<TableReference> {
+        // Manually parse the table reference to avoid case folding.
+        if case_sensitive {
```

Comment:
Might be better to re-use Datafusion implementation?

https://github.com/apache/datafusion/blob/main/datafusion/common/src/utils/mod.rs#L276

```
pub(crate) fn parse_identifiers_normalized(s: &str, ignore_case: bool) -> Vec<String> {
    parse_identifiers(s)
        .unwrap_or_default()
        .into_iter()
        .map(|id| match id.quote_style {
            Some(_) => id.value,
            None if ignore_case => id.value,
            _ => id.value.to_ascii_lowercase(),
        })
        .collect::<Vec<_>>()
}
```